### DISCUSSION TITLE
Is full manual control (with Generacs bypass switch) a possibility?

### DISCUSSION THREAD

**Discussion by skorpioskorpio** [jgyates/genmon#1179] [General]
So here is my issue (sorry for the long post but I think it needs context):

My power rarely just goes off for extended periods, occasionally, but not that often. What it does do however is thrash, on-off-on-off... multiple times over a short period of time, it will also do this as an after effect of an extended outage as well. This happens a lot, like maybe 20-30 times a year. I've had my generator for a couple years now (Generac 26kW, air cooled Evo 2), and I think it's only actually started because of a utility loss, maybe twice. The auto-start timer is set to the minimum, but as I understand it the 'mains failed auto-start' timer and the 'mains are back auto-stop' timer is the same timer... which just seems idiotic to me, but anyway. So what happens? Even if the generator starts, it returns to mains before everything is settled and everything still gets beat up by the pulsing and the generator isn't even finished with it's cool down run yet. Anyway, simply put, it hasn't really solved the problem I bought it to solve.

For the first year I had the vendors annoying cellular modem attached. Which of course when my first year was up they sent me a bill for the 'privileged' of having a device on there that made it so I couldn't do anything else as it used the only port on the generator. I had the pintsize.me board sitting in a box with a Pi waiting for the day. Well, I really wasn't interested in paying a bill for an annoyance, and told them to remove the modem, which apparently meant it also terminated my service contract, which seemed stupid since they never did made any effort to contact me for anything, or for that matter pay attention to any output, but it is what it is I guess.

I finally got motivated enough to go and try and at least join the generator to the built-in WiFi and found that it doesn't exist, not only did they put their stupid cell modem on my unit, but they removed the WiFi that was supposed to come with the generator. Oh WTF! So GenMon to the rescue, wow should have done this way sooner! Everything I had hoped for from the generator and more, it's fantastic. A couple things though:

I've lost 4 TVs and have had to rebuild quite a few computers because of this, not to mention it puts half of my IoT devices into reset pairing mode when one of these on-off-on-off... events happens. So all the TVs and significant electronics are going onto a 6kW UPS wired into the house, but that just solves part of the problem. And yea I always knew that the generator alone won't keep things going without a blip, but if the generator doesn't do what I want it to, it doesn't really solve much of the problem at all. What I want is to have the generator start immediately and stay running until I tell it to turn off. Even if that means I need to be the one to turn it on, or well, via a script. 

I plan to use MQTT through Home Assistant to do this, and plan to have everything necessary to keep that up on UPS. Right now if I put everything I plan to put on the UPS, my run time would be very short, a couple of minutes. But I am going to max out the UPS on external batteries soon (it's an Eaton 9PX6SP BTW) and then it'll have a standby time much longer, like much much longer, more like an hour and a half at full load, which is 30A@240V split phase. But that's only TVs, computers and networking, I still want to auto-start the generator as quickly as possible to avoid the thrashing. 

There are some notes in the docs that make me wonder whether this is possible, specifically, the ones that say you can't send commands to the controller if: "Mains power is disconnected, in any mode" or "The generator is running in 'Manual'", which seems like Plan A and Plan B. Does this mean I am stuck with the whims of the Evo controller's idiotic one size fits all timer? My thought was to set that timer (or if I can't get into the 'dealer mode', pay them to set it) to some comfortably long enough period to keep the generator from switching off and use an automation to start and stop it within that period. 

Can I do that? Or is there some way for me to put the controller on the generator on the UPS as well, so that it never senses a failure, ever? And always thinks the mains are connected even if they aren't (while still using their transfer switch, which is a sub panel switch not a whole house one). Or, probably the costliest solution, put a 100A controllable magnetic contactor in-between the main panel feeder breaker and the sub panel transfer switch, which would control whether the generator thinks there is mains power or not, though this would be my solution of last resort, as such a switch isn't going to be cheap, and not easy to find in split phase.

**Comment by jgyates** [comment]
Hi @skorpioskorpio 

You could try setting the pickup voltage to a higher level. I think the default is 190V.  You can access this value in the dealer menu. You can find the dealer access code [here](https://github.com/jgyates/genmon/wiki/Appendix-M-Tips-and-Tricks-useful-for-Genmon#access-the-dealer-menu). 

Regarding you question, the remote commands only work if the controller is in Auto mode. Also the remote stop command does not work if the generator has been started by a utility outage. The stop command will only stop if the generator was started started by a remote command. I don't know what will happen if you put a long start delay time on the the controller, the tried to start it via remote commands before the start timer expired. If you could do that then there is a possibility that you could control every thing with remote commands but it would need to be tested as the successful functionality of the commands is dependent on the controller.

Info on sending commands via MQTT are here in this thread: https://github.com/jgyates/genmon/issues/932

Let me know if you have any questions or how your testing goes. A feature close to this has been asked before in a few different ways:

[Stop during and outage](https://github.com/jgyates/genmon/issues/846)
[Schedule start when an outage occurs](https://github.com/jgyates/genmon/issues/292)

And the caveats listed above have limited these request.

**Reply by skorpioskorpio** [reply]
It's beyond annoying, what got me remotivated to start looking into this is that this has happened 4 or 5 times just in the last couple weeks. I still find the fact that the timer for determining a loss of mains and the timer to return to mains is the same timer, it's comically stupid, it's kind of rule 1 of generators. I've been dealing with data center UPS/Generator systems for decades, none of them work that way.

**Comment by skorpioskorpio** [comment]
Honestly, if the Generac wasn't so loud, I'd just put a manual magnetic motor starter contactor in-between panels and be done with it. Machine tool contactors are 'no-restart' by design, so that if the power goes out someone doesn't put their hand somewhere potentially dangerous trying to figure out why the machine stopped working. On heavy machinery, loss of power is the same as hitting the kill switch, it actually what the kill switch does, it just unenergizes the contactor relay. But if I did that and it went on when I wasn't home it'd run forever.

**Comment by RalphBSz** [comment]
I know it's months ago, let me add something to the answer above.

As to the on-off-on-off problem, I have no idea; the answer from jgyates is likely the best possible one.

To the question whether the air-cooled Generac with an Evo controller can be remotely controlled (meaning turn it on and have it transfer even when the utility power is on, and conversely have it turn off even when the utility power is off): No, that can't be done in software, as fast as I know. There are no commands you can send the generator over the various serial interfaces that can accomplish that.

The only way to turn the generator into the three Off/Auto/Manual modes is to use the three physical buttons on the control panel. So what I (and a few others) have done: Opened up the Evo controller, added small relays to the wiring harness that goes to the three buttons, and control those relays from the same Raspberry Pi that is running Genmon. That's a little bit non-trivial, since the auto and manual buttons are Normal Open (you can solder a NO relay in parallel to them), but the off button is NC (you have to cut a wire and put the relay's NC contact in series). For my purpose (which is to turn the generator off even when there is no utility power), that is sufficient. For your purpose (turn the generator on and have the transfer switch go to the generator position, even though utility power is on) that may not be enough: I don't know whether the transfer switch will do that; you may have to add one more relay in the transfer switch to fake it into having no utility power. Be very careful: If both coils of the transfer switch are energized at the same time, you can burn them up, which can cause a fire (been there, done that, due to different reasons).

And let me add a set of warnings. Obviously, adding relays to the Evo control board or to the transfer switch is dangerous, and voids any warranty. You also need to think about the generator's motor warming up and cooling down, which needs to be done without an electric load. Since there is no way to turn the transfer switch to either the utility position or to a (non-existing) off position when there is no utility power, that requires adding a large NC contractor to the generators 240V output (with another small relay to drive that contractor). And: if you turn the generator off (by pressing the off button, or with a relay that fakes the off button), the utility coil in the transfer switch is no longer protected against brownouts, and will burn up if a brownout happens (you can guess how I know that). So also add an under voltage protection relay to that coil.

So with a significant amount of extra work, it is possible to achieve what you want done. But it requires much more than connecting a Raspberry Pi and installing some software; it's hours with soldering irons and power wiring.

**Reply by jgyates** [reply]
FYI, the Evolution Liquid Cooled units have remote commands to set Off/Auto/Manual mode. This is available on the Maintenance page, but only on Liquid Cooled Evolution models.
